docs/subnets/create-a-vm-blobvm.md

```diff
@@ -0,0 +1,454 @@
+
+### What are Handlers
+
+Handlers serve the response for the incoming HTTP requests. Handlers can also be wrapped with **gRPC** for efficiently making calls from other services such as avalanchego. Handler helps in creating APIs. VM implements 2 kinds of handlers -
+
+* **Non-Static Handlers** - They help in interacting with blockchains instantiated by the VM. The API's endpoint will be different for different chains. `/ext/bc/[chainID]`
+* **Static Handlers** - They help in directly accessing VM. These are optional for reasons such as parsing genesis bytes required to instantiate new blockchains. `/ext/vm/[vmID]`
+
+### Registering a VM
+
+While registering a VM, we store its **factory** against the user-defined **vmID**. As the name suggests, a VM factory can create new VM instances. Using the VM's instance, we can initialize it and can create a new chain. Note that, instantiating a VM is different than initializing it. We initialize the instance of a VM with parameters like genesis, database manager, etc. to create a functional chain.
+
+```go
+// Registering a factory inside avalanchego
+func (m *manager) RegisterFactory(vmID ids.ID, factory Factory) error {
+    ...
+    m.factories[vmID] = factory
+    ...
+}
+```
+
+### Instantiating a VM
+
+Each VM has a **factory** that is capable of creating new VM instances. A VM can be initialized as a chain along with handlers for accessing it, only when we have the VM's instance. The factory's `New` method shown below, returns the VM's instance to its caller in `avalanchego`.
+
+```go
+// Returning a new VM instance from VM's factory
+func (f *Factory) New(*snow.Context) (interface{}, error) { return &vm.VM{}, nil }
+```
+
+VM's methods and other parts of the code are made available to `avalanchego` by **registering** its factory and instantiating a new VM using this factory whenever a chain is required to be built. It's generally in the [`factory.go`](https://github.com/ava-labs/blobvm/blob/master/factory.go) file of the VM.
+
+```go
+// Instantiating a new VM from its factory in avalanchego
+func (m *manager) buildChain(chainParams ChainParameters, sb Subnet) (*chain, error) {
+    ...
+    vmFactory, err := m.VMManager.GetFactory(vmID)
+    vm, err := vmFactory.New(ctx.Context)
+    ...
+}
+```
+
+### Initializing a VM
+
+Chains are functional, only when the instantiated VMs are initialized. Initializing a VM involves setting up the database, block builder, mempool, genesis state, handlers, etc. This will expose the VM's API handlers, start accepting transactions, gossiping them across the network, building blocks, etc.
+
+```go
+if err := vm.Initialize(
+    ctx.Context,
+    vmDBManager,
+    genesisData,
+    chainConfig.Upgrade,
+    chainConfig.Config,
+    msgChan,
+    fxs,
+    sender,
+);
+```
+
+You can refer to the implementation of `vm.initialize` in the [BlobVM](https://github.com/ava-labs/blobvm/blob/1fbb655246a2a41861f40d603ad40299b177252a/vm/vm.go#L92) repository.
```

Comment:
nit: Unsure if we should use a specific commit version as opposed to master@latest. Both have their pros/cons.
- latest's code is always up-to-date, but the docs will get stale as the vm is updated
- a commit's code is stale, but the docs will always be up-to-date

Reply:
Using master branch.